worker/score.go

```diff
@@ -190,7 +190,6 @@ func (s *ScoreContext) CompareScores(subs []*models.Submission) *models.ProblemR
 			if which == nil {
 				which = sub
 				maxScore = score
-				break
```

Comment:
This is a fairly random change. Why are we changing this?

Reply:
The break statement was ineffective. It was breaking out of the switch statements, which doesn't have any effect. Not sure what it's intention was, but after looking into the logic, I think everything works fine without it

Reply:
Was it trying to `break` from the outer loop instead?

Reply:
Maybe, but then the `if which == nil` serves no purpose.

Reply:
It probably means that we can break the loop right there and stop considering later submissions; without the `break` things still work (because of `if which == nil`) but it's less efficient...